packages/react-query/src/internals/createRootHooks.tsx

```diff
@@ -0,0 +1,506 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import {
+  DehydratedState,
+  QueryClient,
+  useInfiniteQuery as __useInfiniteQuery,
+  useMutation as __useMutation,
+  useQueries as __useQueries,
+  useQuery as __useQuery,
+  hashQueryKey,
+  useQueryClient,
+} from '@tanstack/react-query';
+import { TRPCClientErrorLike, createTRPCClient } from '@trpc/client';
+import type { AnyRouter } from '@trpc/server';
+import { Observable } from '@trpc/server/observable';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  CreateClient,
+  TRPCProvider,
+  TRPCQueryOptions,
+  UseDehydratedState,
+  UseTRPCInfiniteQueryOptions,
+  UseTRPCInfiniteQueryResult,
+  UseTRPCMutationOptions,
+  UseTRPCMutationResult,
+  UseTRPCQueryOptions,
+  UseTRPCQueryResult,
+  UseTRPCSubscriptionOptions,
+} from '../shared/hooks/types';
+import { createUseQueriesProxy } from '../shared/proxy/useQueriesProxy';
+import { CreateTRPCReactOptions, UseMutationOverride } from '../shared/types';
+import { SSRState, TRPCContext } from './context';
+import { TRPCContextState } from './deprecated/TRPCContextState';
+import { QueryType, getArrayQueryKey } from './getArrayQueryKey';
+import { getClientArgs } from './getClientArgs';
+import { useHookResult } from './useHookResult';
+import { TRPCUseQueries } from './useQueries';
+
+/**
+ * @internal
+ */
+export function createRootHooks<
```

Comment:
this is the new "root hooks"

Reply:
the contents i've just copy/pasted from `createHooksInternal`, but has no generic shit